packages/html/src/button/button-structure.json

```
@@ -0,0 +1,211 @@
+                "name": "disabled",
+                "selector": [
+                    ".k-disabled"
+                ]
+            }
+        ],
+        "options": [
+            {
+                "name": "size",
+                "values": [
+                    {
+                        "name": "small",
+                        "selector": [
+                            ".k-button-sm"
+                        ]
+                    },
+                    {
+                        "name": "medium",
+                        "selector": [
+                            ".k-button-md"
+                        ]
+                    },
+                    {
+                        "name": "large",
+                        "selector": [
+                            ".k-button-lg"
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "rounded",
+                "values": [
+                    {
+                        "name": "small",
+                        "selector": [
+                            ".k-rounded-sm"
+                        ]
+                    },
+                    {
+                        "name": "medium",
+                        "selector": [
+                            ".k-rounded-md"
+                        ]
+                    },
+                    {
+                        "name": "large",
+                        "selector": [
+                            ".k-rounded-lg"
+                        ]
+                    },
+                    {
+                        "name": "full",
+                        "selector": [
+                            ".k-rounded-full"
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "fillMode",
+                "values": [
+                    {
+                        "name": "solid",
+                        "selector": [
+                            ".k-button-solid"
+                        ]
+                    },
+                    {
+                        "name": "flat",
+                        "selector": [
+                            ".k-button-flat"
+                        ]
+                    },
+                    {
+                        "name": "outline",
+                        "selector": [
+                            ".k-button-outline"
+                        ]
+                    },
+                    {
+                        "name": "link",
+                        "selector": [
+                            ".k-button-link"
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "themeColor",
+                "needs": "fillMode",
+                "values": [
+                    {
+                        "name": "base",
+                        "selector": [
+                            ".k-button-${fillMode}-base"
```

Can be scoped to `options` for clarity.

```suggestion
                            ".k-button-${options.fillMode}-base"
```